net/xia/ppal_ether/main.c

```diff
@@ -81,6 +81,264 @@ static int local_dump_ether(struct fib_xid *fxid, struct fib_xid_table *xtbl,
 	return -EMSGSIZE;
 }
 
+/* ETHER main table operations. */
+static int main_newroute(struct xip_ppal_ctx *ctx, struct fib_xid_table *xtbl,
+			 struct xia_fib_config *cfg)
+{
+	struct interface_addr *neigh_addr, *exist_addr;
+	struct xip_deferred_negdep_flush *dnf;
+	struct fib_xid_ether_main *mether;
+	struct net_device *out_interface;
+	struct fib_xid *cur_fxid;
+	u32 nl_flags, bucket;
+	const char *id;
+	int rc;
+
+	/* Check for errors in cfg. */
+	if (!cfg->xfc_dst || !cfg->xfc_odev || !cfg->xfc_lladdr 
+		        || cfg->xfc_lladdr_len != cfg->xfc_odev->addr_len )
+		return -EINVAL;
+	
+	out_interface = cfg->xfc_odev;
+	nl_flags      = cfg->xfc_nlflags;
+	id            = cfg->xfc_dst->xid_id;
+
+	if (!(out_interface->flags & IFF_UP) || (out_interface->flags & IFF_LOOPBACK))
+		return -EINVAL;
+
+	neigh_addr = allocate_interface_addr(out_interface , cfg->xfc_lladdr , GFP_ATOMIC);
```

Comment:
`main_newroute()` is called in process context since it called via Netlink, so we don't need to use `GFP_ATOMIC` here. Please use `GFP_KERNEL` instead.

Moreover, `allocate_interface_addr()` is only called here, so it doesn't need its last parameter `gfp_t flags`.